boot/bootutil/include/bootutil/fault_injection_hardening.h

```diff
@@ -0,0 +1,343 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2020 Arm Limited
+ */
+
+#ifndef __FAULT_INJECTION_HARDENING_H__
+#define __FAULT_INJECTION_HARDENING_H__
+
+/* Fault injection mitigation library.
+ *
+ * Has support for different measures, which can either be enabled/disabled
+ * separately or by defining one of the MCUBOOT_FIH_PROFILEs.
+ *
+ * NOTE: These constructs against fault injection attacks are not guaranteed to
```

Comment:
There is another PR, which is a test suite on top of QEMU:
https://github.com/JuulLabs-OSS/mcuboot/pull/789

This can be used with any compiler to validate the hardening code. It does not provide a 100% test coverage against FI on the entire mcuboot binary. It testes only that code sections which are between the FIH_CALL_START and FIH_CALL_END macros.